share/rocm/cmake/ROCMSetupVersion.cmake

```diff
@@ -154,7 +154,18 @@ endfunction()
 
 function(rocm_set_soversion LIBRARY_TARGET SOVERSION)
     if(NOT WIN32)
-        rocm_version_regex_parse("^([0-9]+).*" SOVERSION_MAJOR "${SOVERSION}")
-        set_target_properties(${LIBRARY_TARGET} PROPERTIES VERSION ${SOVERSION} SOVERSION ${SOVERSION_MAJOR})
+        rocm_version_regex_parse("^([0-9]+).*" LIB_VERSION_MAJOR "${SOVERSION}")
+        rocm_version_regex_parse("^[0-9]+\\.([0-9]+).*" LIB_VERSION_MINOR "${SOVERSION}")
+
+        if (DEFINED ENV{ROCM_LIBPATCH_VERSION})
+            set (LIB_VERSION_PATCH $ENV{ROCM_LIBPATCH_VERSION})
+        else ()
+            set (LIB_VERSION_PATCH 0)
```

Comment:
If `ROCM_LIBPATCH_VERSION` is not defined then an additional `.` should not be added at all.

Reply:
The idea with the patch was to add ".<patch-ver>" to the soname name always. Eg: if ROCM_LIBPATCH_VERSION is defined then ".20901" or ".0" since default is 0. I believe there is no case where only "." is added. Please clarify on this.